### ISSUE TITLE
Aircraft Spawning

### ISSUE THREAD

**Issue by jamesg31** [jamesg31/TowerGame#3]
Need to create a better way to spawn aircraft. Likely will need to use spawnpoints for gates ect. Also need to create list of airlines and GA aircraft.

**Comment by AdamJCavanaugh** [comment]
I am working on this, some... a debug-mode that will let you select/enter an aircraft name, location, altitude, and speed, then spawn it... hopefully. :)

**Comment by AdamJCavanaugh** [comment]
![image](https://user-images.githubusercontent.com/47431642/112174361-bc14c880-8bcc-11eb-8e05-b2fc2c847ef4.png)
Please forgive the hodgepodge nature of this sketch, but what do you think of something like this for an airplane menu/GUI? We can have it be read-only for Name, but be able to enter anything when you're in debug mode (to spawn a new airplane in a random location)?

**Comment by AdamJCavanaugh** [comment]
And I recognize this might fit better under issue #5 , but as I was considering how to implement spawning aircraft before we have any details set on _where_ to spawn aircraft, this is what I came up with...

**Comment by jamesg31** [comment]
I think we are going to need to add "gates" to the config file, and maybe paths that the aircraft can follow on the ground. These can act as spawn locations for departures. We will need to figure out something else when we implement approaches I guess.

**Comment by AdamJCavanaugh** [comment]
Yeah, for sure... in the meantime, does the mockup work okay for you as an update to the generic aircraft GUI? If not, does it work specifically for adding airplanes in debug mode, I guess? If not, I'll work on something else.

**Comment by jamesg31** [comment]
Oh of course, I get what your asking. It totally works!

**Comment by AdamJCavanaugh** [comment]
Hey, check out [this commit](https://github.com/jamesg31/TowerGame/blob/manual-aircraft-spawn-debug/menu-test.py) (it's standalone of just the GUI) and let me know? Obviously, we can/should add some theming and we'll need to come up with something for scaling... but when you enter stuff, it gets some checks (work needed) and printed (can call a function to spawn/update aircraft)...

**Comment by AdamJCavanaugh** [comment]
I left it out of the code, but since everything "belongs" to top_panel, you can just top_panel.hide() and top_panel.show() to do the whole GUI.

There's the whole "anchors" aspect that... _should_ make resizing/relative work? I just didn't figure it out yet and I'm done for now. :P

**Comment by AdamJCavanaugh** [comment]
@jamesg31 , any thought on the example code snippet? If it's good, I would plan to replace the UI class with this block, basically. But I don't want to start on that larger effort until I get your thoughts on it. :)

**Comment by jamesg31** [comment]
Test

**Comment by jamesg31** [comment]
Sorry, I have had a busy couple days. This looks really good. I'm pretty happy with this implementation.

**Comment by AdamJCavanaugh** [comment]
No time for real-life stuff when we have an unpaid project to work on! 🤣 

When I do things like this, I'm not trying to rush you... just making sure I remember where I am on things. 😄

**Comment by jamesg31** [comment]
No of course! Your work on this is invaluable

**Comment by jamesg31** [comment]
I am going to try and get some work done on this. I am thinking you can define where the gates and taxiways are in a config file, then we can randomly  spawn aircraft at the gates. We can then use a pathfinding algorithm to get the aircraft to taxi to the hold point of the runway, and then depart.
